File: server/src/models/Lesson.ts
import Discipline from './Discipline';
import LessonTime from './LessonTime';
import Model from './Model';
import Room from './Room';
import Teacher from './Teacher';

class Lesson extends Model {
	readonly Discipline: Discipline;

	readonly Time: LessonTime;

	readonly Room: Room;

	readonly Teacher: Teacher;

	constructor(name: string, discipline: Discipline, time: LessonTime, room: Room, teacher: Teacher) {
		super(name);
		this.Discipline = discipline;
		this.Time = time;
		this.Room = room;
		this.Teacher = teacher;
	}
}

export default Lesson;

File: worker-genetic/models/lesson.py
from .discipline import Discipline
from .interval import Interval
from .room import Room
from .teacher import Teacher
from .model import Model


class Lesson(Model):

    def __init__(self, name: str, discipline: Discipline, interval: Interval, room: Room, teacher: Teacher) -> None:
        super().__init__(name)

        self.discipline = discipline
        self.interval = interval
        self.room = room
        self.teacher = teacher
